### PR TITLE
added entities for clubs and categories

### PR DESCRIPTION
**Pull request by IrenZh**

### CODE REVIEW

**Review by janestadnytska — APPROVED** [review]

File: src/main/java/org/ssu/edu/teachua/db/entities/Category.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.ssu.edu.teachua.db.annotation.Column;
+import org.ssu.edu.teachua.db.annotation.ManyToMany;
+import org.ssu.edu.teachua.db.annotation.TableDB;
+
+import java.util.Set;
+
+@TableDB(name = "categories")
+@Data
+@Setter
+@Getter
```
Comment (Maksym637) [inline src/main/java/org/ssu/edu/teachua/db/entities/Category.java:15]: @ Data = Constructor + @ Setter + @ Getter + @ String;

File: src/main/java/org/ssu/edu/teachua/db/entities/Categories.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import lombok.Setter;
+import org.ssu.edu.teachua.db.annotation.Column;
+import org.ssu.edu.teachua.db.annotation.TableDB;
+
+import java.math.BigInteger;
+
+@TableDB(name = "categories")
+@Data
+@Setter
+@Getter
```
[the remark this answers is not in the document]
Comment (IrenZh) [reply src/main/java/org/ssu/edu/teachua/db/entities/Categories.java]: remove, thanks

File: src/main/java/org/ssu/edu/teachua/db/entities/Category.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Column(name = "tag_text_color")
+    public String tagTextColor;
+
+    @Column(name = "url_logo")
+    public String urlLogo;
+
+    @ManyToMany(foreignTable ="clubs", tableForManyToMany = "club_category",
+            mainColumnDB = "category_id", foreignColumnDB = "club_id")
+    private Set<Club> club;
```
Comment (Maksym637) [inline src/main/java/org/ssu/edu/teachua/db/entities/Category.java:38]: image
I don't know if we need @ ManyToMany annotation as @ ManyToMany = @ ManyToOne and @ OneToMany.
But in other way, your implementation is also good. We need clarifications from Liubomyr and  Ilya : )

**Review by Maksym637 — APPROVED** [review]

File: src/main/java/org/ssu/edu/teachua/db/entities/Category.java (hunk 3 above)
Comment (IrenZh) [reply src/main/java/org/ssu/edu/teachua/db/entities/Category.java:38]: Ilya helped me with this annotation, so we have approving by him))))

Comment (Maksym637) [reply src/main/java/org/ssu/edu/teachua/db/entities/Category.java:38]: O, nice !

File: src/main/java/org/ssu/edu/teachua/db/entities/ClubsCategory.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@Data
+public class ClubsCategory extends Entity {
+
+    @Column(name = "club_id")
+    public BigInteger clubId;
+
+    @Column(name = "category_id")
+    public BigInteger categoryId;
+
+}
```
Comment (janestadnytska) [inline src/main/java/org/ssu/edu/teachua/db/entities/ClubsCategory.java:19]: do we need to specify here:
 **@ManyToOne(foreignTable = "clubs", foreignColumnDB = club_id")
    private Club club;**
    or no?

File: src/main/java/org/ssu/edu/teachua/db/repository/impl/ClubsCategoryDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ClubsCategoryDAOImpl extends AbstractDAOImpl<Club> {
+    private static final Map<String, AbstractDAOImpl> entitiesDaoImpl;
+
+    static {
+        entitiesDaoImpl = new TreeMap<>();
+        entitiesDaoImpl.put("categories", new CategoryDAOImpl());
+    }
```
Comment (janestadnytska) [inline src/main/java/org/ssu/edu/teachua/db/repository/impl/ClubsCategoryDAOImpl.java:14]: maybe we need to add here :
**entitiesDaoImpl.put("clubs", new ClubDAOImpl());**

File: src/main/java/org/ssu/edu/teachua/db/repository/impl/CategoryDAOImpl.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CategoryDAOImpl extends AbstractDAOImpl<Category> {
+
+    private static final Map<String, AbstractDAOImpl> entitiesDaoImpl;
+
+    static {
+        entitiesDaoImpl = new TreeMap<>();
+        entitiesDaoImpl.put("clubs", new ClubsCategoryDAOImpl());
```
Comment (lhalam) [inline src/main/java/org/ssu/edu/teachua/db/repository/impl/CategoryDAOImpl.java:14]: `new ClubsCategoryDAOImpl()` =>`new ClubDAOImpl()`

**Pull request merged**